src/components/overview-table.js

```diff
@@ -111,7 +158,60 @@ function FileDataRow({ processFile, fileGroup, timestamp }) {
                 {processFileStatus}
             </TableCell>
             <TableCell data-test={fileType + '-' + fileGroup + '-filename'}>
-                {processFile.filename}
+                {availableInputs !== undefined ? (
+                    <FormControl>
+                        <Select
+                            sx={SELECT_STYLES}
+                            displayEmpty
+                            onChange={handleClickOpen}
+                            value={processFilename}
+                        >
+                            <MenuItem value={processFilename}>
+                                <em>{processFilename}</em>
+                            </MenuItem>
+
+                            {availableInputs
+                                .filter((input) => input.fileType === fileType)
+                                .filter(
+                                    (input) =>
+                                        input.fileName !== processFilename
+                                )
+                                .map((input) => (
+                                    <MenuItem value={input.fileName}>
+                                        {input.fileName}
+                                    </MenuItem>
+                                ))}
+                        </Select>
+                        <Dialog open={open} onClose={handleClose}>
```

Comment:
If possible, extract this `Dialog` in a separate component to simplify `FileDataRow`.